### PR TITLE
fix: (Platform) fix Upload Collection issues

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
closes https://github.com/SAP/fundamental-ngx/issues/4039
closes https://github.com/SAP/fundamental-ngx/issues/4478
closes https://github.com/SAP/fundamental-ngx/issues/4477
closes https://github.com/SAP/fundamental-ngx/issues/4475
closes https://github.com/SAP/fundamental-ngx/issues/4472
closes https://github.com/SAP/fundamental-ngx/issues/4472
#### Please provide a brief summary of this pull request.
Fixed Upload Collection issues
#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [X] update `README.md`
- [X] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

**Comment by puru-hk** [comment]
Tested the below issues and it is working as expected.
 #4039
 #4478
#4477
 #4475
 #4472
 #4472

File: apps/docs/src/app/platform/component-docs/platform-upload-collection/platform-upload-collection-examples/platform-upload-collection-base-data-provider.ts [hunk 1]
```diff
@@ -306,6 +315,8 @@ export class PlatformUploadCollectionDataProviderExample extends UploadCollectio
                     return res;
                 }, {});
 
+                console.log(11111, currentItem.files);
```
Comment (kavya-b) [inline apps/docs/src/app/platform/component-docs/platform-upload-collection/platform-upload-collection-examples/platform-upload-collection-base-data-provider.ts:318]: is this log needed?

**Comment by puru-hk** [comment]
There is one issue. After created the folder try to add the files under folder. Browse and select the files it is not showing the selected filed under folder. Could you please have a look.
Steps:
1. Create a new folder
2. Click on the new folder
3. Add the new files under the folder 
4. Browse and select the files
5. Selected files are not shown under the files

**Review by kavya-b — APPROVED** [review]
Looks good, verified all related issues' fixes work on netlify.

**Review by dimamarksman — APPROVED** [review]

**Review by fkolar — APPROVED** [review]

**Comment by puru-hk** [comment]
Looks good verify the above issue it is working as expected.

**Review by puru-hk — APPROVED** [review]

**Pull request merged**